### PR TITLE
[menu] Group and Separator components

### LINKED ISSUE #70: [Separator] Implement Separator
Create the Separator component that can be used in Menus and Selects.
By default it can be an `<hr />` element with `role=separator`.

**Search keywords**:

### PR DESCRIPTION
**Pull request by michaldudak**
Created the Group and Separator components for the Menu.

Closes #70

### CODE REVIEW

File: packages/mui-base/src/Separator/Root/SeparatorRoot.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * - [SeparatorRoot API](https://base-ui.netlify.app/components/react-separator/#api-reference-SeparatorRoot)
+ */
+const SeparatorRoot = React.forwardRef(function SeparatorRootComponent(
+  props: SeparatorRoot.Props,
+  forwardedRef: React.ForwardedRef<HTMLDivElement>,
+) {
+  const { className, render, ...other } = props;
+
+  const ownerState = {};
```
Comment (atomiks) [inline packages/mui-base/src/Separator/Root/SeparatorRoot.tsx:23]: Should be outside the component

File: packages/mui-base/src/Menu/Group/MenuGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace MenuGroup {
+  export interface Props extends BaseUIComponentProps<'div', OwnerState> {
+    /**
+     * The content of the component.
+     */
+    children?: React.ReactNode;
+  }
+
+  export type OwnerState = {};
```
Comment (atomiks) [inline packages/mui-base/src/Menu/Group/MenuGroup.tsx:64]:
```suggestion
  export interface OwnerState {}
```

File: packages/mui-base/src/Menu/Group/MenuGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ * - [Menu](https://base-ui.netlify.app/components/react-menu/)
+ *
+ * API:
+ *
+ * - [MenuGroup API](https://base-ui.netlify.app/components/react-menu/#api-reference-MenuGroup)
+ */
+const MenuGroup = React.forwardRef(function MenuGroup(
+  props: MenuGroup.Props,
+  forwardedRef: React.ForwardedRef<Element>,
+) {
```
Comment (atomiks) [inline packages/mui-base/src/Menu/Group/MenuGroup.tsx:21]: Can Menu groups have labels?

Comment (michaldudak) [reply packages/mui-base/src/Menu/Group/MenuGroup.tsx:22]: Added `MenuGroupLabel` component.

**Review by atomiks — APPROVED** [review]

**Comment by colmtuite** [comment]
@michaldudak Label needs to be added to the Anatomy section code.

**Review by colmtuite — APPROVED** [review]

**Comment by michaldudak** [comment]
Done.

**Pull request merged**